src/AST/Solved.hs

```diff
@@ -82,7 +89,7 @@ data Exp_ = LNum String
           | LStr String
           | LBool String
           | JSExp String
-          | App Exp Exp
+          | App Exp Exp Bool
```

Comment:
That serves to identify the end of an application. Because the problem is that a parse expression like this:
```
fn(1, 3)(4)
```
builds an AST like this:
```
App (App (App 1 fn) 3) 4
```

So we don't know when compiling where were the parens and since I move the compilation to fn(1, 3)(4) instead of the previous fn(1, 3, 4), I needed that information somehow in the Solved AST, but its source is the Source AST tough, so now we get that AST for the above expression:

```
App (App (App 1 fn False) 3 True) 4 True
```
So, I know that after the application of 1, I need to put a comma, but after 3 ( True for final ), I need to stop the application and eventually start a new one if there are more args to apply.